tensorflow/compiler/jit/mark_for_compilation_pass.cc

```diff
@@ -1010,6 +1010,45 @@ Status MarkForCompilationPassImpl::BuildInitialClusterSet() {
   return Status::OK();
 }
 
+StatusOr<bool> IsIdentityDrivingConstsInLoop(Node* node) {
+  if (!node->IsIdentity()) {
+    return false;
+  }
+
+  // Check if one of the ancestors is a Switch node.
+  Node* switch_node = nullptr;
+  for (const Edge* e : node->in_edges()) {
+    if (e->src()->IsSwitch()) {
```

Comment:
You probably want to ignore control edges?

Reply:
Actually, can't we use `input_node` like we do for `maybe_loopcond` below?

Reply:
On a second thought, let's also check the Identity node is connected to the true-path output of a Switch to make the pattern more robust.